compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1856,7 +1863,56 @@ impl<'a> Resolver<'a> {
                         )
                     },
                 )
-            };
+            });
+
+            // 3. check whether the name refers to an item in local scope
+            // then delay to check whether methond name is an associated function or not
+            // For example:
+            // ```
+            // struct Foo;
+            // impl Foo {
+            //    fn bar(&self) {}
+            // }
+            //
+            // fn main() {
+            //    let foo = Foo {};
+            //    foo::bar(); // suggest to foo.bar();
+            //}
+            //```
+            if suggestion.is_none() &&
+                let Some(ribs) = ribs &&
+                let Some(LexicalScopeBinding::Res(Res::Local(local_id))) = self.resolve_ident_in_lexical_scope(
+                    ident,
+                    ValueNS,
+                    parent_scope,
+                    None,
+                    &ribs[ValueNS],
+                    ignore_binding,
+                )
+                {
+                    let sm = self.session.source_map();
+                    // only check path with two segments, like `foo::bar(..)`
```

Comment:
Is this the same as the `path: &[Segment]` parameter that we pass to this function? Can we just check if it's len is `2` and use its segments to create this span?

Reply:
Nice! the code is much clean now.